src/hooks/useCreateGrant.ts

```diff
@@ -0,0 +1,50 @@
+import { useCallback } from 'react';
+import { useSigner, useAccount } from 'wagmi';
+
+import { functionRequest } from '../supabase';
+
+const domain = {
+  name: 'ENS Grants',
+  version: '1',
+  chainId: 1,
+};
+
+const types = {
+  Grant: [
+    { name: 'address', type: 'address' },
+    { name: 'roundId', type: 'uint256' },
+    { name: 'title', type: 'string' },
+    { name: 'description', type: 'string' },
+    { name: 'fullText', type: 'string' },
+  ],
+};
+
+export function useCreateGrant() {
+  const { data: signer } = useSigner();
+  const { data: account } = useAccount();
+
+  const createGrant = useCallback(
+    async (roundId: number, title: string, description: string, fullText: string) => {
```

Comment:
should `description` & `fullText` both be required?

Reply:
yea, we need a short description to show in the list, but full text for the actual proposal.